include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t              maxClipDistances;
+    //uint32_t              maxCullDistances;
+    //uint32_t              maxCombinedClipAndCullDistances;
+    //uint32_t              discreteQueuePriorities;
+    float pointSizeRange[2];
+    float lineWidthRange[2];
+    //float                 pointSizeGranularity;
+    //float                 lineWidthGranularity;
+    //VkBool32              strictLines;
+    //VkBool32              standardSampleLocations;
+    //VkDeviceSize          optimalBufferCopyOffsetAlignment;
+    //VkDeviceSize          optimalBufferCopyRowPitchAlignment;
+    uint64_t nonCoherentAtomSize;
+            
+    /* Vulkan 1.1 Core  */
+    uint32_t subgroupSize;
+    core::bitflag<asset::IShader::E_SHADER_STAGE> subgroupOpsShaderStages;
+    bool shaderSubgroupBasic = false;
+    bool shaderSubgroupVote = false;
+    bool shaderSubgroupArithmetic = false;
+    bool shaderSubgroupBallot = false;
+    bool shaderSubgroupShuffle = false;
+    bool shaderSubgroupShuffleRelative = false;
+    bool shaderSubgroupClustered = false;
+    bool shaderSubgroupQuad = false;
+    bool shaderSubgroupQuadAllStages = false; //quadOperationsInAllStages;
+    //VkPointClippingBehavior    pointClippingBehavior;
+    //uint32_t                   maxMultiviewViewCount;
+    //uint32_t                   maxMultiviewInstanceIndex;
+    //VkBool32                   protectedNoFault;
+    //uint32_t                   maxPerSetDescriptors;
+    //VkDeviceSize               maxMemoryAllocationSize;
+            
+    /* Vulkan 1.2 Core  */
+    //VkShaderFloatControlsIndependence    denormBehaviorIndependence;
+    //VkShaderFloatControlsIndependence    roundingModeIndependence;
+    //VkBool32                             shaderSignedZeroInfNanPreserveFloat16;
+    //VkBool32                             shaderSignedZeroInfNanPreserveFloat32;
+    //VkBool32                             shaderSignedZeroInfNanPreserveFloat64;
+    //VkBool32                             shaderDenormPreserveFloat16;
+    //VkBool32                             shaderDenormPreserveFloat32;
+    //VkBool32                             shaderDenormPreserveFloat64;
+    //VkBool32                             shaderDenormFlushToZeroFloat16;
+    //VkBool32                             shaderDenormFlushToZeroFloat32;
+    //VkBool32                             shaderDenormFlushToZeroFloat64;
+    //VkBool32                             shaderRoundingModeRTEFloat16;
+    //VkBool32                             shaderRoundingModeRTEFloat32;
+    //VkBool32                             shaderRoundingModeRTEFloat64;
+    //VkBool32                             shaderRoundingModeRTZFloat16;
+    //VkBool32                             shaderRoundingModeRTZFloat32;
+    //VkBool32                             shaderRoundingModeRTZFloat64;
+    //uint32_t                             maxUpdateAfterBindDescriptorsInAllPools;
```

Comment:
in GL its going to be something like either 16k or just `~0u`

Reply:
lets go with `~0u`

Reply:
I want to expose in 2 phases

## Update After Bind and nonUniformEXT shader qualifier

Descriptor Lifetime Tracking PR #345 will do this, cause I don't want to rewrite the tracking system again.

## Actual Descriptor Indexing

The whole 512k descriptor limits, runtime desc arrays, etc. will come later